src/typecheck.c

```diff
@@ -263,6 +267,9 @@ static const Type *check_binop(
     }
     if (got_pointers) {
         cast_type = voidPtrType;
+    }
+    if (got_numbers && got_floats) {
+        cast_type = floatType;

```

Comment:
This logic needs more work IMO. Here's how it should behave, I think:
- `int + float = float`
- `int + double = double`
- `float + float = float`
- `double + double = double`
- `float + double = double`

So use `double` whenever one of the two sides is a double.